SourceCode/chaos_hw_02/chaos_hw_02.cpp

```diff
@@ -0,0 +1,84 @@
+﻿#include <iostream>
+#include <fstream>
+#include <vector>
+#include <cstdlib>
+#include <ctime>
+
+
+void CreateGridImage(const int numRectY, const int numRectX, const int maxColorComponent, const int rectHeight, const int rectWidth, std::vector<std::vector<std::vector<int>>>& image)
+{
+    for (int y = 0; y < numRectY; ++y) {
+        for (int x = 0; x < numRectX; ++x) {
+            std::vector<int> color = { rand() % (maxColorComponent + 1) , rand() % (maxColorComponent + 1) , rand() % (maxColorComponent + 1) };
+            for (int j = 0; j < rectHeight; ++j) {
+                for (int i = 0; i < rectWidth; ++i) {
+                    int pixelX = x * rectWidth + i;
+                    int pixelY = y * rectHeight + j;
+                    switch (rand() % 4) {
+                    case 0: image[pixelY][pixelX] = { color[0] , 0, 0 }; break;
+                    case 1: image[pixelY][pixelX] = { 0, color[1], 0 }; break;
+                    case 2: image[pixelY][pixelX] = { 0, 0, color[2] }; break;
+                    case 3: image[pixelY][pixelX] = color; break;
+                    }
+                }
+            }
+        }
+    }
+}
+
+void CreateCircleImage(const int imageHeight, const int imageWidth, std::vector<std::vector<std::vector<int>>>& image)
+{
+    for (int y = 0; y < imageHeight; ++y) {
+        for (int x = 0; x < imageWidth; ++x) {
+            float distance = std::sqrt(std::pow(x - imageWidth / 2, 2) + std::pow(y - imageHeight / 2, 2));
```

Comment:
It's always better to avoid the expensive sqrt operation, here you could go use the square distances instead.

Reply:
Yes, I super agree with this and I would put more optimization if it would be on GPU, but for c++ I believed common optimalization magic happens 😄